packages/clay-color-picker/src/__tests__/index.tsx

```diff
@@ -467,15 +489,113 @@ describe('Interactions', () => {
 
 			fireEvent.click(dropdownToggle as HTMLButtonElement, {});
 
-			const blankSplotch = editorGetByTitle(
+			const colorSplotch = editorGetByTitle(
 				'008000'
 			) as HTMLButtonElement;
 
-			fireEvent.click(blankSplotch as HTMLButtonElement, {});
+			fireEvent.click(colorSplotch as HTMLButtonElement, {});
 
-			expect(document.body).toMatchSnapshot();
+			const colorEditorToggle = (
+				document.body as HTMLElement
+			).querySelector('.clay-color-header button');
+
+			fireEvent.click(colorEditorToggle as HTMLButtonElement, {});
 
 			expect(input.value).toBe('008000');
+
+			expect(
+				document.body.querySelectorAll('button[title="dfcaff"]').length
+			).toBe(0);
+
+			expect(editorGetByTestId('rInput').value).toBe('0');
+			expect(editorGetByTestId('bInput').value).toBe('0');
+			expect(editorGetByTestId('gInput').value).toBe('128');
+
+			expect(editorGetByTestId('customHexInput').value).toBe('008000');
+		});
+
+		it('only updates the inputs and the color editor components', () => {
+			const input = editorGetByLabelText(
+				/Color selection is/
+			) as HTMLInputElement;
+
+			fireEvent.change(input, {target: {value: 'DFCAFF'}});
+
+			const colorEditorToggle = (
+				document.body as HTMLElement
+			).querySelector('.clay-color-header button');
+
+			fireEvent.click(colorEditorToggle as HTMLButtonElement, {});
+
+			const rInput = editorGetByTestId('rInput');
+			const bInput = editorGetByTestId('bInput');
+			const gInput = editorGetByTestId('gInput');
+
+			fireEvent.change(rInput, {target: {value: '200'}});
+
+			fireEvent.change(gInput, {target: {value: '200'}});
+
+			fireEvent.change(bInput, {target: {value: '200'}});
+
+			expect(input.value).toBe('c8c8c8');
+
+			expect(editorGetByTestId('customHexInput').value).toBe('C8C8C8');
+		});
+
+		it('can duplicate colors', () => {
+			const input = editorGetByLabelText(
+				/Color selection is/
+			) as HTMLInputElement;
+
+			fireEvent.change(input, {target: {value: 'DFCAFF'}});
+
+			const dropdownToggle = document.querySelector('.dropdown-toggle');
+
+			fireEvent.click(dropdownToggle as HTMLButtonElement, {});
+
+			const blankSplotch = (
+				document.body as HTMLButtonElement
+			).querySelector('button[title="FFFFFF"]');
+
+			fireEvent.click(blankSplotch as HTMLButtonElement, {});
+
+			expect(input.value).toBe('DFCAFF');
+
+			const greenSplotch = (
+				document.body as HTMLButtonElement
+			).querySelector('button[title="008000"]');
+
+			fireEvent.click(greenSplotch as HTMLButtonElement, {});
+
+			fireEvent.change(input, {target: {value: 'DFCAFF'}});
+
+			const purpleSplotchs = (
+				document.body as HTMLButtonElement
+			).querySelectorAll('button[title="dfcaff"]');
+
+			expect(purpleSplotchs.length).toBe(2);
+		});
+
+		it('CSS values are displayed in HEX in the custom hex input', () => {
```

Comment:
I think there are two use cases we need to certify, one is named colors like you're doing here, so maybe it's good to define a text that matches that this is testing for named colors and another to check CSS variables with `var(--blue )` for example. I'm not sure if this will work well because maybe in the tests in Jest it's not possible to get the color of the variable, if not we can go without that case.

Reply:
Hey @matuzalemsteles! I've tred using values like `var(--red)` but Jest doesn't provide this kind of input, that's why I used the `red` one. If you want, I can change the definition of the test case.

Reply:
Oh I see, I imagined this I was hoping we would make it 🥲. yes, it is better to change the test definition to understand that we are checking named colors here.